consensus/core/src/block_manager.rs

```diff
@@ -216,10 +216,23 @@ impl BlockManager {
 
     /// Tries to find the provided block_refs in DagState and BlockManager,
     /// and returns missing block refs.
-    pub(crate) fn try_find_blocks(&mut self, mut block_refs: Vec<BlockRef>) -> BTreeSet<BlockRef> {
+    pub(crate) fn try_find_blocks(&mut self, block_refs: Vec<BlockRef>) -> BTreeSet<BlockRef> {
         let _s = monitored_scope("BlockManager::try_find_blocks");
+        let gc_round = self.dag_state.read().gc_round();
+
+        // No need to fetch blocks that are <= gc_round as they won't get processed anyways and they'll get skipped.
+        // So keep only the ones above.
+        let mut block_refs = block_refs
```

Comment:
Havent checked but if checking gc filtered block_refs is something we do often across the codebase, might be worth adding as a method to dag_state?